app/controllers/webhooks_controller.rb

```diff
@@ -15,14 +15,15 @@ def sendgrid
     events.each do |event|
       message_id = Email::MessageIdService.message_id_clean((event["smtp-id"] || ""))
       to_address = event["email"]
+      error_code = event["status"]
       if event["event"] == "bounce"
-        if event["status"]["4."]
-          process_bounce(message_id, to_address, SiteSetting.soft_bounce_score)
+        if error_code[Email::SMTP_STATUS_TRANSIENT_FAILURE]
```

Comment:
I think there might be a potential for a bug here. 

We should be using `error_code.start_with?("4.")` instead of `error_code["4."]` because the latter will match if `error_code == "2.4.1"` 🤔